minicriu-client.c

```diff
@@ -307,4 +336,60 @@ int minicriu_register_new_thread(void) {
 	return 0;
 }
 
+static int getmap() {
+	char line[512];
+	mapscnt = 0;
+	FILE *proc_maps;
+	proc_maps = fopen("/proc/self/maps", "r");
+
+	if (!proc_maps)
+		return 1;
+
+	while (fgets(line, sizeof(line), proc_maps)) {
+		void *addr_start, *addr_end;
+		if (sscanf(line, "%p-%p*", &addr_start, &addr_end) < 2) {
+			fclose(proc_maps);
+			return 1;
+		}
+
+		maps[mapscnt].start = addr_start;
+		maps[mapscnt++].end = addr_end;
+    }
+	fclose(proc_maps);
+
+	return 0;
+}
+
+static int cleanup() {
+	char line[512];
+	FILE *proc_maps;
+	proc_maps = fopen("/proc/self/maps", "r");
+
+	if (!proc_maps)
+		return 1;
+
+	while (fgets(line, sizeof(line), proc_maps)) {
+		void *addr_start, *addr_end;
+		if (sscanf(line, "%p-%p", &addr_start, &addr_end) < 2) {
+			fclose(proc_maps);
+			return 1;
+        }
+
+		int diff = 1;
+		for(int i = 0; i < mapscnt; i++) {
+			if (addr_start == maps[i].start) {
+				diff = 0;
+				break;
+			}
+		}
+
+		if (diff) {
+			if (munmap(addr_start, addr_end - addr_start)) {
+				perror("munmap");
+			}
+		}
```

Comment:
Do we need O(n^2) at all? Is not sufficient just do something like
```
munmap(0, maps[i].start)
for(int i = 1; i < mapscnt; i++) {
  munmap(maps[i-1].end, maps[i].start - maps[i-1].end);
}
munmap(maps[mapcnt-1].end, 0x100.... - maps[mapcnt-1].end);
```
I do expect some troubles with the first and last mmap, probably they only can be adjusted regarding current mapping (but I'd avoid reading the whole /proc/self/maps anyway)